hack/generate-jobs.py

```diff
@@ -0,0 +1,146 @@
+            if errs:
+                errors.extend(errs)
+    return errors
+
+
+def generate_one(path: pathlib.Path,
+                 only_verify: bool, overwrite: bool) -> typing.List[str]:
+    """
+    Generate job configuration files from one template.
+    Return list of errors.
+    """
+    config = configparser.ConfigParser()
+    config.read_file(path.open())
+
+    template_name = config.get("DEFAULT", "template")
+    template_path = path.parent / template_name
+    errors = []
+    with template_path.open() as inp:
+        template = jinja2.Template(inp.read(), lstrip_blocks=True)
+        pairs = config.get("DEFAULT", "files").split(",")
+        for name, job in [pair.split(":") for pair in pairs]:
+            tmp = tempfile.NamedTemporaryFile(
+                "w",
+                prefix=f"{template_name}.",
+                delete=False,
+            )
+            with tmp:
+                header = (
+                    "# GENERATED FILE - DO NOT EDIT!\n#\n# "
+                    f"Instead, modify {template_name} and run `make generate-jobs`.\n"
+                )
+                for section in config.sections():
+                    tmp.write(
+                        template.render(
+                            config[section],
+                            file=name,
+                            job_name=job.format(section=section),
+                            header=header,
+                        )
+                    )
+                    header = ""
+
+            out = template_path.parent / f"{template_path.stem}-{name}.yaml"
+            if not os.path.exists(out):
+                if only_verify:
+                    os.unlink(tmp.name)
+                    errors.append(f"Can't verify content: {out} doesn't exist")
+                    continue
+            else:
+                equal = filecmp.cmp(out, tmp.name, shallow=False)
+                if only_verify:
+                    os.unlink(tmp.name)
+                    if not equal:
+                        errors.append(
+                            f"Generated content for {out} differs from existing"
+                        )
+                    continue
+                if equal:
+                    os.unlink(tmp.name)
+                    continue
+                if not overwrite:
+                    os.unlink(tmp.name)
+                    errors.append(
+                        f"Generated content for {out} differs from existing, "
+                        "use --overwrite to update"
+                    )
+                    continue
+            shutil.move(tmp.name, out)
+
+    return errors
+
+
+def main(argv):
+    """Entry point."""
+    parser = argparse.ArgumentParser(
+        prog="Jobs Generator",
+        description="Generate job configuration files from templates",
+    )
+    parser.add_argument(
+        "pattern",
+        nargs="+",
+        help="config path pattern, e.g config/jobs/kubernetes/sig-node/*.conf",
```

Comment:
Documenting the new support for `**` would be useful here.

Reply:
done, here is how it looks now:
```
$ python hack/generate-jobs.py --help
usage: Jobs Generator [-h] [--only-verify] [--overwrite] pattern [pattern ...]

Generate job configuration files from templates

positional arguments:
  pattern        config path pattern in the Python pathlib pattern language format:
                 https://docs.python.org/3/library/pathlib.html#pattern-language,
                 for example: config/jobs/**/*.generate.conf

options:
  -h, --help     show this help message and exit
  --only-verify  Only verify if generated files are the same as existing
  --overwrite    Owerwrite output files
```

Reply:
What happens when neither `--only-verify` nor `--overwrite` are given?

It looks to me like a single flag should be enough, with overwriting done as default action. I also would rename `--only-verify` to `--verify` - shorter and doesn't make it sound like "only" verifying is somehow inferior.

Reply:
> What happens when neither --only-verify nor --overwrite are given?

Script would generate output files or fail if generated content differs from existing. I thought that defaulting to rewriting existing files could be harmful. As for `--only-verify`,  `only` means that content would be generated and verified, but not written. Without `only` user might think that `--verify` is an additional option for the default action.

All that said, I understand that one option is better than two and open to suggestions.

Reply:
ok, done.